gossip/neighbor.c

```diff
@@ -77,16 +77,14 @@ retcode_t neighbor_send(node_t *const node, neighbor_t *const neighbor,
   if (packet == NULL) {
     return RC_NEIGHBOR_NULL_PACKET;
   }
-  if ((ret = get_transaction_to_request(node->requester, &hash))) {
+  // TODO randomly select milestone or regular tx
+  if ((ret = get_transaction_to_request(node->requester, hash, false))) {
     return ret;
   }
-  if (hash != NULL) {
-    // TODO(thibault): iota_packet_set_request
-    flex_trits_to_trits(hash_trits, NUM_TRITS_HASH, hash->trits,
-                        hash->num_trits, hash->num_trits);
-    trits_to_bytes(hash_trits, packet->content + PACKET_TX_SIZE,
-                   NUM_TRITS_HASH);
-  }
+  // TODO(thibault): iota_packet_set_request
+  flex_trits_to_trits(hash_trits, HASH_LENGTH_TRIT, hash, HASH_LENGTH_TRIT,
```

Comment:
what if hash is the "null hash" ?

Reply:
If it's the null hash, it means that there are no transaction left to be requested but we still need to send that packet so I guess null hash is the way to go... ?